.github/workflows/update-citation.yml

```diff
@@ -0,0 +1,39 @@
+name: Update CITATION.cff file when contributors.yaml changed
+
+on:
+  push:
+    branches: "master"
+    paths:
+      - contributors.yaml
+
+jobs:
+  update-citation:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          lfs: false
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Update citation file
+        run: tox -e citation
+      - name: Check whether the citation metadata from CITATION.cff is valid
+        uses: citation-file-format/cffconvert-github-action@2.0.0
+        with:
+          args: "--validate"
+      - name: Push updated citation file back to repository
+        run:
+          git config --local user.name github-actions[bot]
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          if ! git diff --exit-code; then
+            git add CITATION.cff
+            git commit -m "Updating CITATION.cff file"
+            git push -f origin master
```

Comment:
1.  master is protected branch. Do we need to update the protections somehow? Here's a [_long_ discussion](https://github.com/orgs/community/discussions/25305).
2.  Why does it need `-f`?

Reply:
The force push shouldn't be required here I was just not careful enough in checking when I adapted code in [this answer from another discussion](https://github.com/orgs/community/discussions/25234#discussioncomment-4026272).

Yeah branch protections could be problematic here, and thinking about it automatically creating a pull request may be a more sensible approach, or just adding a step to the checks which runs update script and checks if any changes produced (or set up a corresponding pre-commit hook).

Reply:
I've now removed this workflow in favour of just adding an option to just check for differences when running `update_citation.py` script and added this to the `tox` `check` environment. This seems a safer solution and still should achieve the goal of making sure citation file is kept up to date if contributor data is changed.